### ISSUE TITLE
setting header changes value of ngx.header['content-type']

### ISSUE THREAD

**Issue by leafo** [openresty/openresty#92]
Not sure if this is a bug, but it's pretty unexpected behavior.

nginx version: openresty/1.7.10.1

``` nginx
worker_processes 1;
error_log stderr notice;
daemon off;
pid logs/nginx.pid;

events {
  worker_connections 1024;
}

http {
  include mime.types;

  server {
    listen 8080;

    location /set/ {
      default_type text/html;
      content_by_lua '
        ngx.header["x-hello"] = "world"
        ngx.print(ngx.header["content-type"])
      ';
    }

    location /noset/ {
      default_type text/html;
      content_by_lua '
        ngx.print(ngx.header["content-type"])
      ';
    }
  }
}
```

Not setting a header, ngx.header["content-type"] is nil

``` bash
> curl -i http://localhost:8080/noset/hello.png
HTTP/1.1 200 OK
Server: openresty/1.7.10.1
Date: Fri, 27 Mar 2015 20:34:58 GMT
Content-Type: image/png
Transfer-Encoding: chunked
Connection: keep-alive

nil
```

Setting a header, ngx.header["content-type"] gets the value presumably from mime.types and the extension.

``` bash
> curl -i http://localhost:8080/set/hello.png
HTTP/1.1 200 OK
Server: openresty/1.7.10.1
Date: Fri, 27 Mar 2015 20:35:05 GMT
Content-Type: image/png
Transfer-Encoding: chunked
Connection: keep-alive
x-hello: world

image/png
```

**Comment by agentzh** [comment]
@leafo Hmm, this is not really a bug though it deserves better treatment. The default `Content-Type` header value is set lazily (upon the first header sending operation or the first response header setting). I think maybe we should also configure the default type upon first response header read to avoid this surprise. Will you consider contributing a patch to ngx_lua? Thanks!

**Comment by zot24** [comment]
Just want to confirm that this is still happening on `openresty/1.9.7.4`

**Comment by agentzh** [comment]
@zot24 It's not been taken care of. The latest OpenResty 1.11.2.2 has nothing changed here.

@membphis @moonbingbing Will you take care of this for me? ;)

**Comment by moonbingbing** [comment]
@agentzh ok, we will try

**Comment by agentzh** [comment]
@leafo Just fixed through openresty/lua-nginx-module#936. Thanks @moonbingbing for the patch.
